### ISSUE TITLE
feature request: extract TRIGSTRs

### ISSUE THREAD

**Issue by Cokemonkey11** [wurstscript/wurst4vscode#40]
Would be useful for migrating old maps and it should be fairly trivial to do, I would think?

**Comment by Frotty** [comment]
do you mean just plain extract? why not use an mpq editor?

**Comment by Cokemonkey11** [comment]
I meant to extract and replace "TRIGSTR_003" in wurst, with contents of map strings. (Strings get created by GUI AFAIK)

**Comment by Frotty** [comment]
Already done for wurst created objects. Inlining of WTS entries is more a task for the finalizer, not compiler imo.

**Comment by Cokemonkey11** [comment]
Hmm, I was thinking just a tool in the editor. But extracting to a file would be fine too. I don't care too much how it's done.

**Comment by Frotty** [comment]
I still don't get it.
> I meant to extract and replace "TRIGSTR_003" in wurst

In wurst? What does that even mean?

> Hmm, I was thinking just a tool in the editor

So a command just to extract the .wts from a map? Why?

Maybe write a user story of what you're actually trying to accomplish. Seems pretty pointless to me.

**Comment by peq** [comment]
I think the story is something like this: 

1. have map written in GUI
2. convert to Jass files
3. replace TRIGSTRs with new tool
4. start rewriting Jass to Jurst/Wurst
5. ???
6. profit

**Comment by Frotty** [comment]
Ah okay, GUI.
I was confused because nonGUI map code only contains like 4 TRIGSTRs.
But who converts GUI maps to wurst?

**Comment by Cokemonkey11** [comment]
> But who converts GUI maps to wurst?

I did, here: https://www.hiveworkshop.com/pastebin/575155029c53ff3ff4364546b25ed4bc16336/

It was for the contest, which had a theme "Update a map that is 15 years old".

In this case it was particularly annoying because the starting map was a cinematic with hundreds of TRIGSTRs

**Comment by Krypt0n** [comment]
use this: https://bin.wurstlang.org/dipejosobi.wurst

**Comment by Cokemonkey11** [comment]
Just from skimming that code, it looks like it won't work if you have two TRIGSTR on the same line

**Comment by Krypt0n** [comment]
that is true,  I used it only once, may need adaptions to other situations

**Comment by Frotty** [comment]
mpqs can now be extracted to folder or single files extract - right from within vscode.
